src/target/windows/mod.rs

```diff
@@ -0,0 +1,61 @@
+use std::mem;
+use winapi::shared::minwindef::FALSE;
+use winapi::um::processthreadsapi::{
+    CreateProcessW, OpenProcess, PROCESS_INFORMATION, STARTUPINFOW,
+};
+use winapi::um::winbase;
+use winapi::um::winnt;
+
+/// This structure holds the state of the debuggee on windows systems.
+pub struct Target {
+    process_handle: winnt::HANDLE,
+}
+
+macro_rules! wide_string {
+    ($string:expr) => {{
+        use std::os::windows::ffi::OsStrExt;
+        let input = std::ffi::OsStr::new($string);
+        let vec: Vec<u16> = input.encode_wide().chain(Some(0)).collect();
+        vec
+    }};
+}
+
+impl Target {
+    /// Launch a new debuggee process.
+    pub fn launch(path: &str) -> Result<Target, Box<dyn std::error::Error>> {
+        let mut si: STARTUPINFOW = unsafe { mem::zeroed() };
```

Comment:
I'd suggest using [MaybeUninit](https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html) here to explicitly tell the compiler that this value is not initialised and is not to be used.

So it can be used as e.g.
```rust
let mut si: STARTUPINFOW = MaybeUninit::uninit();
let mut pi: PROCESS_INFORMATION = MaybeUninit::uninit();

CreateProcessW(
    ...
    si.as_mut_ptr(),
    pi.as_mut_ptr(),
);
```

Reply:
`STARTUPINFOW` is an input value, not an output value.

https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessw

> The caller is responsible for ensuring that the standard handle fields in STARTUPINFO contain valid handle values.

Reply:
Makes sense then, thanks!

Reply:
Shouldn't i use the `zeroed` function of `MaybeUninit`?
This
```rust
let mut si: STARTUPINFOW = mem::MaybeUninit::uninit();
```
be more like
```rust
let mut si= mem::MaybeUninit::<STARTUPINFOW>::zeroed();
let mut si = unsafe{si.assume_init()};
```
`mem::MaybeUninit::uninit()` requires us to `drop` explicitly.

Reply:
The first member of `STARTUPINFO` is the size of the whole struct.